services/registration/src/routes/application.ts

```diff
@@ -224,6 +225,26 @@ applicationRouter.route("/:id/actions/submit-application").post(
       throw new BadRequestError("Unable to submit application data. Branch is not currently open.");
     }
 
+    const applicantType = branch.applicationGroup;
+    const autoConfirm = branch.automaticConfirmation;
+
+    if (
+      autoConfirm?.enabled &&
```

Comment:
Additionally, you'll need to check if the branch type is application, as automatic confirmation is only applicable on `APPLICATION` branches.